okhttp/src/main/java/okhttp3/internal/http2/Http2Codec.java

```diff
@@ -90,13 +90,17 @@
   final StreamAllocation streamAllocation;
   private final Http2Connection connection;
   private Http2Stream stream;
+  private final boolean useH2C;
```

Comment:
nit: seems clearer to just assign the Protocol here (HTTP_2 or H2C) and just pass in as a param to readHttp2HeadersList, instead of setting HTTP_2 and then based on this boolean overriding the value.